src/NuGet.Clients/VsExtension/VisualStudioAccountProvider.cs

```diff
@@ -0,0 +1,390 @@
+                // throw so that the other providers will not be called
+                throw new Exception("could not get the account manager, unable to check the keychain");
+            }
+
+            VSAccountProvider provider = null;
+            provider = (VSAccountProvider) await _accountManager
+                .GetAccountProviderAsync(VSAccountProvider.AccountProviderIdentifier).ConfigureAwait(false);
+
+            if (provider == null)
+            {
+                // we know this is a VSO endpoint but we are unable to get the credentials so we should
+                // throw so that the other providers will not be called
+                throw new Exception("could not get the account provider, unable to check the keychain");
+            }
+
+            //  Ask keychain for all accounts
+            var accounts = _accountManager.Store.GetAllAccounts();
+
+            //  Look through the accounts to see what ones have the VSO tenant in them (collected from
+            //  the LookupTenant() call at the top of the method).
+            foreach (var account in accounts)
+            {
+                var tenant = FindTenantInAccount(account, uriTenantId, provider);
+                if (tenant != null)
+                {
+                    posibleAccounts.Add(new AccountAndTenant(account, tenant));
+                }
+            }
+
+            if (posibleAccounts.Count == 1)
+            {
+                //  If we only have one posible account use it
+                ret = await GetTokenFromAccount(posibleAccounts[0], provider, nonInteractive, cancellationToken)
+                    .ConfigureAwait(false);
+                if (isRetry)
+                {
+                    var hasAccess = await AccountHasAccess(uri, proxy, ret, cancellationToken)
+                        .ConfigureAwait(false);
+                    if (!hasAccess)
+                    {
+                        // The account didn't have access and we are on a retry so the token didn't expire.
+                        // we either need to prompt the user for different creds or fail in this case
+                        ret = await PromptUserForAccount(uriTenantId, provider, nonInteractive, cancellationToken)
+                            .ConfigureAwait(false);
+                    }
+                }
+            }
+            else if (posibleAccounts.Count > 1)
+            {
+                var accountsWithAccess = new List<AccountWithCreds>();
+                foreach (var account in posibleAccounts)
+                {
+                    var cred = await GetTokenFromAccount(
+                        account,
+                        provider,
+                        nonInteractive: true,
+                        cancellationToken:cancellationToken)
+                        .ConfigureAwait(false);
+
+                    var hasAccess = await AccountHasAccess(uri, proxy, cred, cancellationToken)
+                        .ConfigureAwait(false);
```

Comment:
If we're absolutely certain that all upstream callers are NEVER calling this in a blocking fashion, ie: Get(...).GetAwaiter().GetResult(), then it's safe to remove the ConfigureAwait(false). If not, we'll deadlock, because SynchronizationContext.Current will be available and set, which will cause a deadlock when execution attempts to resume on the calling thread.